compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -8587,8 +8594,12 @@ TR::Node *idivSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
       {
       int32_t divisor = secondChild->getInt();
       int32_t shftAmnt = -1;
-      if (divisor != 0)
+      if (divisor != 0
+          && performTransformation(s->comp(), "%sConstant non-zero divisor for idiv allows any parent DIVCHK to be removed\n", s->optDetailString()))
```

Comment:
> Also, do you think it would be good to limit the `performTransformation()` to cases where the parent is a `DIVCHK`?

I want to make sure I understand what you're suggesting.  Were you thinking I might create a function like this?

```
static bool contextIsOKForDivisionOrRemainderSimplification(TR::Simplifier s, TR::Node *node)
   {
   return (_curTree->getNode()->getOpCodeValue() != TR::DIVCHK
           || _curTree->getNode()->getFirstChild() != node
           || performTransformation(s->comp(), "%sConstant non-zero divisor allows parent DIVCHK [%p] to be removed\n", _curTree->getNode())));
   }
```
and my new calls to `performTransformation` could become:
```      
if (divisor != 0 && contextIsOKForDivisionOrRemainderSimplification(s, node))
```

I suppose further that I could move the assignment `s->_nodeToDivchk = NULL;` into that new function to isolate it.

Reply:
Yes, that looks great. That way we won't get transformation messages about removing a `DIVCHK` that isn't there, and we won't dump the trees just because simplifier saw some division like `x / 5`

Reply:
Factored these changes into the new function `permitSimplificationOfConstantDivisor` in commit 04b2b82964f525e4f3c4d7c88a7fb0ec9d1a1b21